apache/Dockerfile

```diff
@@ -97,9 +98,9 @@ RUN set -ex; \
     \
     export GNUPGHOME="$(mktemp -d)"; \
     export GPGKEY="3D06A59ECE730EB71B511C17CE752F178259BD92"; \
-    curl -fsSL -o phpMyAdmin.tar.xz $URL; \
-    curl -fsSL -o phpMyAdmin.tar.xz.asc $URL.asc; \
-    echo "$SHA256 *phpMyAdmin.tar.xz" | sha256sum -c -; \
+    curl -fsSL -o phpMyAdmin.tar.xz "https://files.phpmyadmin.net/phpMyAdmin/5.2.1/phpMyAdmin-5.2.1-all-languages.tar.xz"; \
+    curl -fsSL -o phpMyAdmin.tar.xz.asc "https://files.phpmyadmin.net/phpMyAdmin/5.2.1/phpMyAdmin-5.2.1-all-languages.tar.xz.asc"; \
+    echo "373f9599dfbd96d6fe75316d5dad189e68c305f297edf42377db9dd6b41b2557 *phpMyAdmin.tar.xz" | sha256sum -c -; \
```

Comment:
this should remain using the ENVs (`ENV`) from the Dockerfile

Reply:
IMO this is bad practice.

The build process is strongly tied to the version being built, as we've just recently seen with b0305371a48c72b78f36b34105cb161758e3366b. A Dockerfile that was created to build v5.2.1 isn't necessarily compatible with v5.2.0, not to say with v4.0.0. If one wants to build the container of a previous version, one must use the Dockerfile matching this version, i.e. from the commit of this repo that was tagged with the requested version. Otherwise the build will either fail (best case), cause unexpected issues at runtime (like a broken config), or even cause security issues.

Reply:
Well, let's let @ibennetch comment about how it would impact maintenance between versions.

But for me this needs to be reverted, since the ENVs are there. They need to be used.

> The build process is strongly tied to the version being built

Well, that's not really true. You can most probably fire most major versions and it will work
phpMyAdmin is more tied to php and extensions than to a specific build

> Dockerfile that was created to build v5.2.1 isn't necessarily compatible with v5.2.0, not to say with v4.0.0

Yeah but this repository only does support for the latest supported versions.
So we dump out the specific build and refresh it.
There is no point for us on this repository to support older versions.

Reply:
> But for me this needs to be reverted, since the ENVs are there. They need to be used.

Why do they need to be used? :confused:

Don't confuse ENVs with ARGs (build arguments). Environment variables are meant to be used *inside* the container. Even though phpMyAdmin doesn't use these env variables, users might add custom software to the container requiring to know the phpMyAdmin version being used. That's why it's common practice to add an env variable with the primary software's version. However, usually they are not called `VERSION`, but `<software>_VERSION`. Adding a `SHA256` env variable is not so common though. I just didn't change that because it's a possible BC-breaking change for software inside the container relying on these env variables.

As explained, you're rather using the ENVs like build arguments. Build arguments (ARG directive) were created to affect the build process (like changing the software's version), i.e. how you're using the ENVs right now. However, one must not use the ARG directive for official images - for exactly the reason I've explained, among others.

> > The build process is strongly tied to the version being built
> 
> Well, that's not really true. You can most probably fire most major versions and it will work phpMyAdmin is more tied to php and extensions than to a specific build

Even though a Dockerfile **might** work with any other phpMyAdmin, there's no guarantee that it works - unless you're willing to give this guarantee. Like, are you confident that I can build with today's Dockerfile the latest phpMyAdmin version as of today in 10 years? If you want to support the latest version only, do exactly this - i.e. bake the version into the Dockerfile. This change is not about supporting older versions. It's the exact opposite: Explicitly not supporting it.